lib/src/sql/id.rs

```diff
@@ -106,6 +107,16 @@ impl From<Vec<Value>> for Id {
 	}
 }
 
+impl From<Number> for Id {
+	fn from(v: Number) -> Self {
+		match v {
+			Number::Int(v) => v.into(),
+			Number::Float(v) => v.to_string().into(),
+			Number::Decimal(v) => v.to_string().into(),
+		}
+	}
+}
+
```

Comment:
I _was_ a bit concerned (see earlier comment) that converting floats and decimals to strings here could be a foot-gun as it facilitates a situation in which `record:1` (where 1 is `Id::Number` and `record:1` (where 1 is `Id::String`) are different things.

After investigating, I found that `Id::String` consisting of a number gets escaped with angle brackets. Since this makes the issue more visible, I'm no longer as concerned.

Reply:
Let's discuss this somewhere else for later! We also need to discuss `Number`s as a whole. Lots of improvements to be made there!